src/macros.rs

```diff
@@ -0,0 +1,48 @@
+/// format a given string with the passed variables
+/// This macro is creating an single used Hashmap, for performance optimizations it might be
+/// more efficient to reuse an existing one.
+///
+/// # Arguments
+/// * `inst` - A string with an Rust-style format instructions
+/// * `values` - A list of values to use for formatting
+///
+/// # Errors
+/// see [strfmt]
+///
+/// # Example
+/// ```
+/// use strfmt::FmtError;
+/// use strfmt::{strfmt,strfmt_builder};
+///
+/// let first = "test";
+/// //test  77.65
+/// println!("{}",strfmt!("{first}{second:7.2}", first,second => 77.6543210).unwrap());
```

Comment:
Curious about the use of `=>` vs `=`. Can we use the latter? If not, why not? I'm not remembering what is allowed in macros.

Reply:
I was curious and tried your suggestion, but it's not possible as it counts as an assignment expression.

However, even if it were possible, I would not want to reuse the `=`. It makes the expression harder to read and can lead to misunderstandings, especially since macros generate code that is not directly visible.